internal/checker/terraform/rules/aws/verify_aws_api_gateway_endpoints_uses_certificate_for_authentication.rego

```diff
@@ -0,0 +1,89 @@
+# METADATA
+# title: "Verify AWS API Gateway endpoints uses client certificate authentication"
+# description: "This policy is designed to ensure the secure authentication of Amazon Web Services (AWS) API Gateway endpoints by verifying that client certificate authentication is implemented."
+# scope: package
+# related_resources:
+# - https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/api_gateway_stage
+# custom:
+#   id: CB_TFAWS_368
+#   severity: MEDIUM
+package lib.terraform.CB_TFAWS_368
+
+isvalid(block) {
+	block.Type == "resource"
+	block.Labels[_] == "aws_api_gateway_stage"
+}
+
+resource[resource] {
+	block := pass[_]
+	resource := concat(".", block.Labels)
+}
+
+resource[resource] {
+	block := fail[_]
+	resource := concat(".", block.Labels)
+}
+
+pass[block] {
+	block := input[_]
+	isvalid(block)
+	block.Labels[_] == "aws_api_gateway_stage"
+	block.Attributes.client_certificate_id != ""
+}
+
+pass[block] {
+	block := input[_]
+	isvalid(block)
+	block.Labels[_] == "aws_apigatewayv2_stage"
+	block.Attributes.client_certificate_id != ""
+}
+
+getApigatewayv2ApiLabel[label] {
+	block := input[_]
+	block.Type == "resource"
+	block.Labels[_] == "aws_apigatewayv2_api"
```

Comment:
add this line here after checking the labels
block.Attributes.protocol_type != "WEBSOCKET"

then you can remove "isApigatewayv2ApiValidType" function